readium/streamer/src/main/java/org/readium/r2/streamer/parser/epub/EpubDeobfuscator.kt

```diff
@@ -76,7 +82,17 @@ internal class EpubDeobfuscator(
     }
 
     private fun getHashKeyAdobe(pubId: String) =
-        pubId.replace("urn:uuid:", "")
-            .replace("-", "")
-            .toHexBytes()
+        toHexBytes(
+            pubId.replace("urn:uuid:", "")
+                .replace("-", "")
+        )
+
+    private fun toHexBytes(string: String): ByteArray? {
+        // Only even-length strings can be converted to an Hex byte array, otherwise it crashes
+        // with StringIndexOutOfBoundsException.
+        if (string.isEmpty() || string.length % 2 != 0) {
+            return null
+        }
+        return string.toHexBytes()
+    }
 }
```

Comment:
In my understanding, only hexadecimal characters can be converted into an hex byte array, so I would rather check in the caller that the parameter looks like an UUID.

Reply:
It's an UUID (allegedly?) for Adobe but a SHA-1 digest for IDPF. If we can find an official spec for Adobe I would be okay being more strict. Until then, just checking that we have a non-empty UUID and an even number of characters sounds safer to me.

Reply:
I suggest to add this check only for the Adobe case, as a SHA-1 hash looks exactly like an UUID . Look at the implementation of the `toHexBytes` method. How would you encode the letter z? It is not valid in radix 16, so the result of `Character.digit("z", 16)` would be -1, which doesn't make any sense.

Reply:
> This is a good point, we can check that it contains only characters from 0 to f.
That's exactly what I mean, I don't care about the ID having "-" or not, we remove them anyway.

Reply:
Oh, I was confused by French Wikipedia which claims that an UUID version 5 is a SHA-1 hash. Here is the explanation:
> Version-5 UUIDs are similar, but SHA-1 is used instead of MD5. Since SHA-1 generates 160-bit digests, the digest is truncated to 128 bits before the version and variant bits are replaced.